server/src/main/java/org/apache/iotdb/db/engine/merge/task/MergeMultiChunkTask.java

```diff
@@ -328,18 +262,57 @@ private void pathsMergeOneFile(int seqFileIdx, IPointReader[] unseqReaders)
     }
   }
 
-  private List<Integer> filterNoDataPaths(List[] seqChunkMeta, int seqFileIdx) {
-    // if the last seqFile does not contains this series but the unseqFiles do, data of this
-    // series should also be written into a new chunk
-    List<Integer> ret = new ArrayList<>();
-    for (int i = 0; i < currMergingPaths.size(); i++) {
-      if ((seqChunkMeta[i] == null || seqChunkMeta[i].isEmpty())
-          && !(seqFileIdx + 1 == resource.getSeqFiles().size() && currTimeValuePairs[i] != null)) {
-        continue;
+  private void readChunkMetaAndModifications(
+      PreviewIterator<Pair<String, List<ChunkMetadata>>> measurementChunkMetadataListIterator,
+      List<Modification>[] modifications,
+      List<ChunkMetadata>[] seqChunkMeta,
+      TsFileResource currTsFile,
+      String deviceId) {
+    if (!measurementChunkMetadataListIterator.hasNext()) {
+      return;
+    }
+    Pair<String, List<ChunkMetadata>> nextFileMeasurement =
+        measurementChunkMetadataListIterator.previewNext();
+    boolean noMoreMeasurement = false;
+    int i = 0;
+    while (!noMoreMeasurement) {
+      String currMergingMeasurement = currMergingPaths.get(i).getMeasurement();
+      if (currMergingMeasurement.equals(nextFileMeasurement.left)) {
+        // current series is found in the file
+        modifications[i] = resource.getModifications(currTsFile, currMergingPaths.get(i));
+        seqChunkMeta[i] = nextFileMeasurement.right;
+        modifyChunkMetaData(seqChunkMeta[i], modifications[i]);
+        for (ChunkMetadata chunkMetadata : seqChunkMeta[i]) {
+          resource.updateStartTime(currTsFile, deviceId, chunkMetadata.getStartTime());
+          resource.updateEndTime(currTsFile, deviceId, chunkMetadata.getEndTime());
+        }
```

Comment:
Use chunkMetadata to update resource may be uncorrect. For example, the time range of chunkMetadata is 0-100, but the data in 10-100 has been deleted and there is no data within this time range in unseq files,  so actually the time range of this device is 0~10.

Reply:
Yes, but it remains a superset of the actual time range so there is no correctness problem (it will not miss any possible query results). Moreover, the ChunkMetadata has been previously inserted with modifications; if possible, a more specific time range should be provided thereafter.

Reply:
got it.